packages/gatsby/src/internal-plugins/query-runner/page-query-runner.js

```diff
@@ -45,11 +49,25 @@ const runQueries = async () => {
   const cleanIds = findIdsWithoutDataDependencies()
 
   // Construct paths for all queries to run
-  const pathnamesToRun = _.uniq([
-    ...runQueriesForPathnamesQueue,
-    ...dirtyIds,
-    ...cleanIds,
-  ])
+  let pathnamesToRun = _.uniq([...dirtyIds, ...cleanIds])
+
+  // If this is the initial run, remove pathnames from `runQueriesForPathnamesQueue`
+  // if they're also not in the dirtyIds or cleanIds.
+  //
+  // We do this because the page component reducer/machine always
+  // adds pages to runQueriesForPathnamesQueue but during bootstrap
+  // we may not want to run those page queries if their data hasn't
+  // changed since the last time we ran Gatsby.
```

Comment:
This introduces regression:
`gatsby build` -> edit query -> `gatsby build` - we will not run query, but query did change

Reply:
Fixed here https://github.com/gatsbyjs/gatsby/pull/11897/commits/5100484904723f30e2d4816f15645be822ac9f42

Thanks for identifying this!